### ISSUE TITLE
Menu window size is erroneously clamped to main viewport size

### ISSUE THREAD

**Issue by arnaud-neny** [ocornut/imgui#7730]
### Version/Branch of Dear ImGui:

Version 1.90.8, Branch: docking

### Back-ends:

imgui_impl_win32.cpp + imgui_impl_dx12.cpp

### Compiler, OS:

Windows 10 + MSVC 2022

### Full config/build information:

```
Dear ImGui 1.90.8 (19080)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 2, sizeof(ImDrawVert): 20
define: __cplusplus=199711
define: _WIN32
define: _WIN64
define: _MSC_VER=1940
define: _MSVC_LANG=201402
define: IMGUI_HAS_VIEWPORT
define: IMGUI_HAS_DOCK
--------------------------------
io.BackendPlatformName: imgui_impl_win32
io.BackendRendererName: imgui_impl_dx12
io.ConfigFlags: 0x00000443
 NavEnableKeyboard
 NavEnableGamepad
 DockingEnable
 ViewportsEnable
io.ConfigViewportsNoDecoration
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x00001C0E
 HasMouseCursors
 HasSetMousePos
 PlatformHasViewports
 HasMouseHoveredViewport
 RendererHasVtxOffset
 RendererHasViewports
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 120.00,11.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

### Details:

Hi!

There is an issue related only to the docking branch with the function CalcWindowAutoFitSize
It's always using the main viewport to define the size_max in some cases. One at least is happening here:
- pop out the "Dear ImGui Demo" window
- resize the main window until it has the smallest size possible
- open "Menu" in the "Dear ImGui Demo"
- go to any menu item with context such "Open Recent"

Thank you!

### Screenshots/Video:

![imgui](https://github.com/ocornut/imgui/assets/11290167/8c551ddb-0141-422f-9c12-0e06fc23df2e)

**Comment by arnaud-neny** [comment]
This issue has been introduced in the 1.90.3

**Comment by ocornut** [comment]
Related to #7287, #7063

**Comment by ocornut** [comment]
I think the fix may be to use:
```cpp
        const bool is_child_not_popup = (window->Flags & ImGuiWindowFlags_ChildWindow) && !(window->Flags & ImGuiWindowFlags_Popup);
        ImVec2 size_min = CalcWindowMinSize(window);
        ImVec2 size_max = (window->ViewportOwned || is_child_not_popup) ? ImVec2(FLT_MAX, FLT_MAX) : ImGui::GetMainViewport()->WorkSize - style.DisplaySafeAreaPadding * 2.0f;
        const int monitor_idx = window->ViewportAllowPlatformMonitorExtend;
        if (monitor_idx >= 0 && monitor_idx < g.PlatformIO.Monitors.Size && !is_child_not_popup)
            size_max = g.PlatformIO.Monitors[monitor_idx].WorkSize - style.DisplaySafeAreaPadding * 2.0f;
```
But I would need to add more tests for the past cases were stumbled into before validating this is good. It's currently too complex/subtle to understand the reasons behind that logic.

**Comment by ocornut** [comment]
Should be fixed by a028c2d. Thanks for reporting!
